### ISSUE TITLE
Question about two times backward?

### ISSUE THREAD

**Issue by pingguokiller** [SNAS-Series/SNAS-Series#10]
Thanks for your publication of the code! This paper is very interesting and I have a question.

![loss](https://user-images.githubusercontent.com/10582481/114525015-502bfa00-9c78-11eb-8fe6-a1e852e03441.png)

As loss = resource_loss.clone() + error_loss.clone(),  why does the "loss " have to backward again after the backward of "resource_loss" and "error_loss"?

In my opinion, one-time‘s backward of "resource_loss" and "error_loss" is necessary.

**Comment by skhu101** [comment]
Thanks for your comment! Here we record the log_alpha's gradient information for the analysis purpose. You can comment out these lines to save the running time.
